Feat/UI translations

SirZemar opened the pull request:
As we spoke before, events component was handling the logic of the translation. Changed that so now only app component does it. 

However, each component has to import the TranslateService from from ngx-translate/core. Is that ok?

SirZemar committed:
feat: language select

SirZemar committed:
refactor(prettier): add scss

SirZemar committed:
add selected value dynamically

SirZemar committed:
fix: change translate property to public

SirZemar commented:
To avoid missing tokens, I believe all the languages folders should follow the same model

SirZemar committed:
import TranslateModule for lazy loading

vdias38 commented:
- Remove lang selector, we'll use browser en as default and set browser lang (`const lang = this.translateService.getBrowserLang();`).
- Remove default angular tokens

SirZemar committed:
refactor: browser language as default

vdias38 commented on package.json:
```diff
@@ -68,6 +70,6 @@
   "lint-staged": {
     "*.ts": "eslint --cache --fix",
-    "*.{ts,scss,html,md}": "prettier --write"
+    "*.{ts,scss,css,md}": "prettier --write"
```
why remove html file type on prettier processing ?

vdias38 commented on src/app/app.component.ts:
```diff
@@ -1,10 +1,16 @@
+    translate.addLangs(['en', 'fr', 'pt']);
+    // translate.setDefaultLang('en');
+    const browserLang = translate.getBrowserLang();
+    translate.use(browserLang.match(/en|pt|fr/) ? browserLang : 'en');
```
how does it work if lang is 'pt-br' ?
pt-br (langCode-region)
should get only langCode, w/o region

vdias38 replied:
use setDefaultLang instead of ternary check

SirZemar committed:
refactor: replace with setDefaultLang

Replace an unnecessary ternary check with a default language in case there are no tokens for the browser language in question

SirZemar commented:
<h4>Started by rebasing main into the branch.</h4>
```
       $git rebase main
```
<h4>On this rebase process it was necessary to solve conflicts since the first commit of the branch (feat: language select #7691f67).
After solving each commit conflict I add all changes and kept going with rebase process.</h4>
```
	$git add .
	$git rebase —continue
```

<img width="1280" alt="Screenshot 2022-06-29 at 16 19 15" src="https://user-images.githubusercontent.com/87383586/176707522-b88917f7-925d-491c-97db-f2cbf32ec346.png">

<img width="1280" alt="Screenshot 2022-06-29 at 16 20 49" src="https://user-images.githubusercontent.com/87383586/176707673-18bbd428-a513-4b9e-be10-63cec3282b33.png">

<h4>When rebase was complete a sync between local and remote repository was made. </h4>

<img width="463" alt="Joses-MacBook-Pro Eventure josemarinho$ git add" src="https://user-images.githubusercontent.com/87383586/176707763-736b5883-e6e2-4d5b-ac3a-892b683e3804.png">

<h4>When rebase was finished, the branch wasn’t in sync with the last master commit, for reason I didn’t understand. It is sync with the commit #55cab36 and it should be with #533e752. </h4>

<img width="1248" alt="Screenshot 2022-06-29 at 16 33 56" src="https://user-images.githubusercontent.com/87383586/176707825-37be1a8f-546a-4c95-bbc8-d854fdc16366.png">

<h4>After rebase it is asked to merge local branch with remote repository, which feels an unnecessary commit.</h4>

<img width="1280" alt="Screenshot 2022-06-29 at 17 05 30" src="https://user-images.githubusercontent.com/87383586/176707912-a3892592-5474-4fa8-81a8-291048374a63.png">

SirZemar commented:
Do I proceed by pushing these changes to remote repository?

vdias38 commented:
I didn't understand what happens, I suggest to do on a pair session a full PR submit workflow to check any issue.
I've redo a PR from current main https://github.com/devizem/Eventure/pull/60.
We can cancel this one.
